login_server/src/cpp/JSONInterface/JsonUnsecureLogin.cpp

```diff
@@ -98,14 +100,25 @@ Poco::JSON::Object* JsonUnsecureLogin::handle(Poco::Dynamic::Var params)
 		result->set("state", "disabled");
 		result->set("msg", "user is disabled");
 		break;
+	case USER_NO_GROUP: 
+		user_model->setGroupId(1);
+		user_model->updateIntoDB("group_id", 1);
+		infos.add("set user.group_id to default group_id = 1");
```

Comment:
no `break` ?

Reply:
that's right, no break
In for most user_state simply a error message will be set, breaked, session will be deleted and returned. 

For USER_NO_PRIVATE_KEY, USER_COMPLETE and USER_EMAIL_NOT_ACTIVATED
a state will be set to success, session_id and user added to return json and info if not empty and then directly return without break. 
For USER_NO_GROUP a default group will be set and saved in db, a info will be added for frontend 
and then go on with set state to success, set user and session_id, set info and return like with user_state USER_NO_PRIVATE_KEY, USER_COMPLETE or USER_EMAIL_NOT_ACTIVATED.

```cpp
switch (user_state) {
	case USER_EMPTY:
	case USER_PASSWORD_INCORRECT:
		result->set("state", "error");
		result->set("msg", "password incorrect");
		break;
	case USER_PASSWORD_ENCRYPTION_IN_PROCESS:
		result->set("state", "processing");
		result->set("msg", "password encryption in process");
		break;
	case USER_KEYS_DONT_MATCH:
		result->set("state", "error");
		result->set("msg", "saved keys mismatch");
		break;
	case USER_DISABLED:
		result->set("state", "disabled");
		result->set("msg", "user is disabled");
		break;
	case USER_NO_GROUP: 
		user_model->setGroupId(1);
		user_model->updateIntoDB("group_id", 1);
		infos.add("set user.group_id to default group_id = 1");
	case USER_NO_PRIVATE_KEY:
	case USER_COMPLETE:
	case USER_EMAIL_NOT_ACTIVATED:
		result->set("state", "success");
		result->set("user", session->getNewUser()->getJson());
		result->set("session_id", session->getHandle());
		session->setClientIp(mClientIP);
		if(infos.size() > 0) {
			result->set("info", infos);
		}
		return result;
	default: 
		result->set("state", "error");
		result->set("msg", "unknown user state");
		result->set("details", USER_NO_GROUP);
	}
	
	sm->releaseSession(session);
	
	return result;
``` 

So for user_state USER_NO_GROUP 
this code will run:
```cpp
user_model->setGroupId(1);
user_model->updateIntoDB("group_id", 1);
infos.add("set user.group_id to default group_id = 1");
result->set("state", "success");
result->set("user", session->getNewUser()->getJson());
result->set("session_id", session->getHandle());
session->setClientIp(mClientIP);
if(infos.size() > 0) {
	result->set("info", infos);
}
return result;
```

Reply:
I would approve this - tho I believe this way is very error prune. If you forget to add a break above the usergroup thing it will result in a successful return + the group_id code execution.

Reply:
Yes that's right, but the whole user group block is only temporary until the new frontend can handle this.